Source/Scene/UrlTemplateImageryProvider.js

```diff
@@ -429,12 +440,72 @@ define([
          */
         hasAlphaChannel : {
             get : function() {
+                //>>includeStart('debug', pragmas.debug);
+                if (!this.ready) {
+                    throw new DeveloperError('hasAlphaChannel must not be called before the imagery provider is ready.');
+                }
+                //>>includeEnd('debug');
                 return this._hasAlphaChannel;
             }
         }
     });
 
     /**
+     * Reinitializes this instance.  This method is primarily intended for use
+     * with the `deferReadiness` flag to the constructor.  Reinitializing an instance already in use is supported, but it is not
+     * recommended because existing tiles provided by the imagery provider will not be updated.
+     * @param {Object} options Any of the options that may be passed to the {@see UrlTemplateImageryProvider} constructor.
+     */
+    UrlTemplateImageryProvider.prototype.reinitialize = function(options) {
```

Comment:
Mention this and the new constructor options in CHANGES.md.

Also, it is probably worth adding an `@example` here to the  reference doc so users have a better idea how to delay initialization.